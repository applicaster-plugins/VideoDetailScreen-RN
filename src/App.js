/* eslint camelcase: 0 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { SafeAreaViewBackwardCompatible } from '@applicaster/london-rn-components';
import VideoDetail from './components/VideoDetail';
import Loader from './components/Loader';
import Header from './components/Header';
import {
  initZappConfig,
  getVideoEntry,
  getRelatedFeed,
  layout,
  parameters
} from './utils';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      collapseText: false,
      relatedLoading: false,
      loading: true
    };
    this.config = initZappConfig(props).catch(e => console.log(e));
    this.handleVideoChange = this.handleVideoChange.bind(this);
    this.toggleCollapseText = this.toggleCollapseText.bind(this);
  }

  componentDidMount() {
    this.config
      .then(() => getVideoEntry(this.props))
      .then(this.handleVideoChange)
      .catch(e => {
        console.log(e);
      });
  }

  async handleVideoChange(toEntry) {
    this.setState({
      entry: toEntry,
      loading: false,
      related: null,
      relatedLoading: true
    });
    const { relatedUrlScheme } = parameters;
    if (!relatedUrlScheme) return;
    try {
      const related = await getRelatedFeed(toEntry);
      this.setState({ related, relatedLoading: false });
    } catch (e) {
      console.log(e);
    }
  }

  toggleCollapseText() {
    this.setState(({ collapseText }) => ({ collapseText: !collapseText }));
  }

  renderVideoDetail() {
    const { toggleCollapseText, handleVideoChange } = this;
    const { entry, related, relatedLoading, collapseText } = this.state;
    return (
      <VideoDetail
        key="detail"
        {...{
          entry,
          related,
          relatedLoading,
          collapseText,
          handleVideoChange,
          toggleCollapseText
        }}
      />
    );
  }

  render() {
    const { loading } = this.state;
    return [
      loading && <Loader key="loader" />,
      !loading && [
        layout.isFullScreen && (
          <SafeAreaViewBackwardCompatible key="container">
            <Header />
            {this.renderVideoDetail()}
          </SafeAreaViewBackwardCompatible>
        ),
        !layout.isFullScreen && this.renderVideoDetail()
      ]
    ];
  }
}

App.propTypes = {
  extra_props: PropTypes.object.isRequired,
  is_tablet: PropTypes.bool
};

export default App;
