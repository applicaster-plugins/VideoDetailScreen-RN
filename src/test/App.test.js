import React from 'react';
import { create } from 'react-test-renderer';
import App from '../App';

describe('App', () => {
  it('renders app', () => {
    const TextInputComponent = create(<App />).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
  });
});
