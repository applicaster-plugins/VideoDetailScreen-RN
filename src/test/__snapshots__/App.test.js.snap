// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders app 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Hello World!
  </Text>
</View>
`;
